### PR TITLE
[64196] - Create TermsOfUseAgreement model

### PR DESCRIPTION
**Pull request by rileyanderson**
## Summary
- Create `TermsOfUseAgreement` model. Migrations will be split out into a [separate PR](https://github.com/department-of-veterans-affairs/vets-api/pull/13632)

## Related issue(s)
- department-of-veterans-affairs/va.gov-team#64196
- department-of-veterans-affairs/vets-api#13632

## How to test
- Checkout and run migrations from here - department-of-veterans-affairs/vets-api#13632
  - Migrate database - `rails db:migrate`
- In rails console create an agreement from a user_account. - `rails c`
  ```ruby
   UserAccount.last.terms_of_use_agreements.create(agreement_version: 'v1', response: :declined)
   UserAccount.last.terms_of_use_agreements.create(agreement_version: 'v1', response: :accepted)
  ```
- The user should have 2 TermsOfUseAgreements, one declined and another accepted 

  ```ruby
   UserAccount.last.terms_of_use_agreements
  ```

## What areas of the site does it impact?
Terms of use

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.

### CODE REVIEW

**Review by asg5704 — APPROVED** [review]

File: spec/factories/terms_of_use_agreements.rb
Comment (asg5704) [inline spec/factories/terms_of_use_agreements.rb:6]: Not a huge deal since this is a spec but I think SuS wanted version 3 and on the FE we would display version 1

**Review by asg5704 — APPROVED** [review]

**Comment by dickdavis** [comment]
LGTM--can't approve yet, though.

I didn't see your original migrations before they were merged, so I did want to see what was your reasoning for going with an enum over a boolean here. Do you anticipate potential values beyond 'accepted' and 'declined'?

Side note: [Rails 7 includes support for native PG enums](https://github.com/rails/rails/pull/41469), which is pretty handy in certain cases. However, I think 0's and 1's are suitable for this particular use case, though that makes me wonder if this field was better off as a boolean.

File: app/models/terms_of_use_agreement.rb
Comment (joeniquette) [inline app/models/terms_of_use_agreement.rb:7]: Testing passes however when a response is submitted the created_at and updated_at are the same time every time I submit a new response. In future versions of this we will need an audit of all submissions to show when a person accepted and denied every time they do.

**Review by joeniquette — APPROVED** [review]
just a question, depending on the answer you can make changes or merge this version in. Functionally it does what the initial version needs to do.

File: app/models/terms_of_use_agreement.rb
Comment (rileyanderson) [reply app/models/terms_of_use_agreement.rb:7]: It will create a new record for every response so a user could potentially have multiple records. The latest record will be used as their current response

**Pull request merged**
